### ISSUE TITLE
Make prefer-template more customizable

### ISSUE THREAD

**Issue by kewisch** [eslint/eslint#6620]
**What version of ESLint are you using?** 2.13.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:** `"prefer-template": 2`

**The problem you want to solve**
I would like to use the [prefer-template](http://eslint.org/docs/rules/prefer-template) rule to make some of our code prettier. I would like to be able to control under which situations template strings are preferred, because I believe there are cases where template strings do not actually improve readability.

Example:
```javascript
// OK, Just a simple append, do not prefer template strings
dump(getComplicatedError() + "\n");

// OK, Just a simple prepend, do not prefer template strings
dump("An error occurred: "  + getComplicatedError());

// Error, expansion inbetween. Prefer template strings!
dump("An error occurred: " + getComplicatedError() + "\n");  
```

**Your take on the correct solution to problem.**
I would suggest to add keyword options, in my example I would use
```json
"prefer-template": [2, { "start": false, "end": false, "inner": true }]
```
All arguments would default to true, hence the shortcut `"prefer-template": 2` would act just the same as before.

**Comment by platinumazure** [comment]
Hi @kewisch, thanks for the issue (and for following the proposed rule change guidelines)! This seems like a potentially worthwhile addition. We can accept this issue once someone on the ESLint team champions this rule and three other people endorse the change (with no one voting against it).

I'm not sure about the option names-- they might need to be tweaked. To me, it's not immediately clear if "start" refers to a string being prepended to a computation at the start of the overall string, or a computation prepended to a string. I think the option names could be made clearer to reflect the order. (In other words, I'm saying that for `var str = "literal" + computation();`, I can't tell whether I should use start or end if I want to add an exemption for this style from prefer-template.)

Hopefully we can get more feedback from the rest of the team as they see this issue.

**Comment by alberto** [comment]
I don't this this is a general enough concern to deserve an option, but you can always create a custom rule tailored to your needs

**Comment by kewisch** [comment]
Looks like this feature is not very well received within the team. I still think this is very useful for the general public because it allows customizing the rule, but of course I am aware that I could write my own rule (as with anything). What is the further process? Are there more team members that would vote on this, or should the issue be closed?

**Comment by vitorbal** [comment]
@kewisch I don't believe there's a high chance this will be accepted into the core, in my opinion, especially since I don't think there are any popular style guides out there that advocate for this amount of granularity when it comes to template literals (please correct me if I'm wrong).

@eslint/eslint-team does anyone feel strongly in favor of this enhancement and would like to champion it?

**Comment by nzakas** [comment]
We have two :-1:s from the team, so I'm going to close this issue. As always, creating a custom rule to do just what you want is an option.
